projects/igniteui-angular/src/lib/data-operations/filtering-strategy.ts

```diff
@@ -74,16 +95,42 @@ export abstract class BaseFilteringStrategy implements IFilteringStrategy  {
                 return true;
             } else {
                 const expression = expressions;
-                const column = grid && grid.getColumnByName(expression.fieldName);
-                const isDate = column ? column.dataType === DateType || column.dataType === DateTimeType : false;
-                const isTime = column ? column.dataType === TimeType : false;
+                let dataType = null;
+                if (!entity) {
+                    const column = grid && grid.getColumnByName(expression.fieldName);
+                    dataType = column?.dataType;
+                } else if (grid.type === 'hierarchical') {
+                    const schema = (grid as IgxHierarchicalGridComponent).schema;
+                    const entityMatch = this.findEntityByName(schema, 'entity');
```

Comment:
I believe it should be just `entity`.
```suggestion
                    const entityMatch = this.findEntityByName(schema, entity);
```